src/components/cylc/analysis/TimeSeries.vue

```diff
@@ -0,0 +1,459 @@
+-->
+
+<template>
+  <Teleport
+    v-if="sortInputTeleportTarget"
+    :to="sortInputTeleportTarget"
+  >
+    <div class="d-flex flex-grow-1 col-gap-1">
+      <v-autocomplete
+        multiple
+        chips
+        closable-chips
+        clearable
+        placeholder="Search"
+        :items="tasks"
+        v-model="displayedTasks"
+        label="Select tasks"
+        ref="selectTasks"
+        @update:search="updateSelectionOptions"
+      >
+        <template
+          v-slot:prepend-item
+        >
+          <v-list-item
+            @click="selectSearchResults"
+          >
+            Select all search results
+          </v-list-item>
+          <v-list-item
+            @click="deselectSearchResults"
+          >
+            Remove all search results
+          </v-list-item>
+          <v-divider class="mt-2"></v-divider>
+        </template>
+      </v-autocomplete>
+      <v-btn
+        @click="jobsQuery(displayedTasks)"
+        data-cy="analysis-jobs-refresh-btn"
+        icon
+        variant="text"
+      >
+        <v-icon :icon="$options.icons.mdiRefresh" />
+        <v-tooltip>Refresh data</v-tooltip>
+      </v-btn>
+      <v-checkbox
+        class="ma-0 pa-0"
+        v-model="showOrigin"
+        label="Show origin"
+        density="compact"
+        hide-details="true"
+        style="min-width: 120px;"
+      />
+    </div>
+  </Teleport>
+  <div id="mainTimeSeries">
+    <VueApexCharts
+      type="line"
+      :options="chartOptions"
+      :series="series"
+      :height="450"
+      width="95%"
+      class="d-flex justify-center"
+          />
+  </div>
+  <div id="miniTimeSeries">
+    <VueApexCharts
+      type="line"
+      :options="miniChartOptions"
+      :series="series"
+      height="120"
+      width="95%"
+      @selection="zoomMainChart"
+      class="d-flex justify-center"
+    />
+  </div>
+  <!-- <p>{{ displayedTasks }}</p> -->
+  </template>
+
+<script>
+import VueApexCharts from 'vue3-apexcharts'
+import {
+  debounce,
+  difference,
+  pick,
+  union,
+  uniq,
+  upperFirst
+} from 'lodash'
+import gql from 'graphql-tag'
+import { formatDuration } from '@/utils/tasks'
+import {
+  mdiDownload,
+  mdiRefresh,
+} from '@mdi/js'
+import { useReducedAnimation } from '@/composables/localStorage'
+
+/** List of fields to request for task for each task */
+const jobFields = [
+  'name',
+  'id',
+  'platform',
+  'cyclePoint',
+  'totalTime',
+  'queueTime',
+  'runTime',
+  'startedTime'
+]
+
+/** The one-off query which retrieves historical job timing statistics */
+const JOB_QUERY = gql`
+query analysisJobQuery ($workflows: [ID], $tasks: [ID]) {
+  jobs(live: false, workflows: $workflows, tasks: $tasks) {
+    ${jobFields.join('\n')}
+  }
+}
+`
+
+/** The callback which gets called when data comes in from the job query */
+class AnalysisJobCallback {
+  /**
+   * @param {Object[]} jobs
+   */
+  constructor (jobs) {
+    this.jobs = jobs
```

Comment:
If extending `DeltasCallback`
```suggestion
    super()
    this.jobs = jobs
```